torchmetrics/functional/text/wer.py

```diff
@@ -12,48 +12,103 @@
+import torch
+from torch import Tensor, tensor
 
-if _JIWER_AVAILABLE:
-    from jiwer import compute_measures
 
+def _edit_distance(prediction_tokens: List[str], reference_tokens: List[str]) -> int:
+    """Standard dynamic programming algorithm to compute the edit distance.
 
-def wer(
+    Args:
+        prediction_tokens: A tokenized predicted sentence
+        reference_tokens: A tokenized reference sentence
+
+    Returns:
+        (int) Edit distance between the predicted sentence and the reference sentence
+    """
+    dp = [[0] * (len(reference_tokens) + 1) for _ in range(len(prediction_tokens) + 1)]
+    for i in range(len(prediction_tokens) + 1):
+        dp[i][0] = i
+    for j in range(len(reference_tokens) + 1):
+        dp[0][j] = j
+    for i in range(1, len(prediction_tokens) + 1):
+        for j in range(1, len(reference_tokens) + 1):
+            if prediction_tokens[i - 1] == reference_tokens[j - 1]:
+                dp[i][j] = dp[i - 1][j - 1]
+            else:
+                dp[i][j] = min(dp[i - 1][j], dp[i][j - 1], dp[i - 1][j - 1]) + 1
+    return dp[-1][-1]
+
+
+def _wer_update(
+    predictions: Union[str, List[str]],
     references: Union[str, List[str]],
+) -> Tuple[Tensor, Tensor]:
+    """Update the wer score with the current set of references and predictions.
+
+    Args:
+        predictions: Transcription(s) to score as a string or list of strings
+        references: Reference(s) for each speech input as a string or list of strings
+
+    Returns:
+        (Tensor) Number of edit operations to get from the reference to the prediction, summed over all samples
+        (Tensor) Number of words over all references
+    """
+    if isinstance(predictions, str):
+        predictions = [predictions]
+    if isinstance(references, str):
+        references = [references]
+    errors = tensor(0, dtype=torch.float)
+    total = tensor(0, dtype=torch.float)
+    for prediction, reference in zip(predictions, references):
+        prediction_tokens = prediction.split()
+        reference_tokens = reference.split()
+        errors += _edit_distance(prediction_tokens, reference_tokens)
+        total += len(reference_tokens)
+    return errors, total
+
+
+def _wer_compute(errors: Tensor, total: Tensor) -> Tensor:
+    """Compute the word error rate.
+
+    Args:
+        errors: Number of edit operations to get from the reference to the prediction, summed over all samples
+        total: Number of words over all references
+
+    Returns:
+        (Tensor) Word error rate
+    """
+    return errors / total
+
+
+def wer(
     predictions: Union[str, List[str]],
+    references: Union[str, List[str]],
     concatenate_texts: bool = False,
-) -> float:
-    """Word error rate (WER_) is a common metric of the performance of an automatic speech recognition system. This
+) -> Tensor:
+    """Word error rate (WER) is a common metric of the performance of an automatic speech recognition system. This
```

Comment:
this makes sure it gets rendered as a link based on this database:
https://github.com/PyTorchLightning/metrics/blob/00f0256db37406df09c18ddc18f4eca99ef7d59e/docs/source/links.rst#L31
```suggestion
    """Word error rate (WER_) is a common metric of the performance of an automatic speech recognition system. This
```